acpi/src/lib.rs

```diff
@@ -49,36 +49,41 @@
 
 #![no_std]
 #![deny(unsafe_op_in_unsafe_fn)]
+#![cfg_attr(feature = "alloc", feature(allocator_api, ptr_as_uninit))]
 
-extern crate alloc;
 #[cfg_attr(test, macro_use)]
 #[cfg(test)]
 extern crate std;
 
+pub mod address;
 pub mod bgrt;
 pub mod fadt;
 pub mod hpet;
 pub mod madt;
 pub mod mcfg;
-pub mod platform;
 pub mod sdt;
 
-pub use crate::{
-    fadt::PowerProfile,
-    hpet::HpetInfo,
-    madt::MadtError,
-    mcfg::PciConfigRegions,
-    platform::{interrupt::InterruptModel, PlatformInfo},
-};
+pub use crate::{fadt::PowerProfile, hpet::HpetInfo, madt::MadtError};
 pub use rsdp::{
     handler::{AcpiHandler, PhysicalMapping},
     RsdpError,
 };
 
+#[cfg(feature = "alloc")]
+pub use crate::{
+    mcfg::PciConfigRegions,
+    platform::{interrupt::InterruptModel, PlatformInfo},
+};
+#[cfg(feature = "alloc")]
+pub mod platform;
+
+#[cfg(feature = "alloc")]
+mod managed_slice;
+#[cfg(feature = "alloc")]
+pub use managed_slice::*;
```

Comment:
Should we re-export this?

Reply:
Not sure what you mean by this, sorry. It was only put in its own separate module to help separate the more complicated memory-code from the rest of the acpi-specific code.

Reply:
Sure, I just meant did we want to make `ManagedSlice` publicly available as `acpi::ManagedSlice` rather than through the submodule, but thinking again I'm fine with it.